### PR TITLE
chore: remove action-button variant property

### PR DESCRIPTION
**Pull request by rubencarvalho**
## Description

Removed the `sp-action-button` `variant` property along with all associated references and dependencies.

## Related issue(s)

- Task: [SWC-302](https://jira.corp.adobe.com/browse/SWC-302)
- Subtask: [SWC-445](https://jira.corp.adobe.com/browse/SWC-445)

## Motivation and context

For the upcoming 1.0.0 release of Spectrum Web Components, we will remove the deprecated components and features.

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [x] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

File: packages/action-button/stories/action-button-white-quiet.stories.ts [hunk 1]
```diff
@@ -20,40 +20,40 @@ export default {
...
-const variant = 'white';
+const static = 'white';
 const quiet = true;
 
 export const XS = (args: Properties): TemplateResult => renderButtons(args);
 XS.args = {
     size: 'xs',
     quiet,
-    variant,
+    static,
```
Comment (Rajdeepc) [inline packages/action-button/stories/action-button-white-quiet.stories.ts:30]:
```suggestion
    staticProp,
```

File: packages/action-button/stories/action-button-white-quiet.stories.ts [hunk 2]
```diff
@@ -20,40 +20,40 @@ export default {
     decorators: [makeOverBackground()],
 };
 
-const variant = 'white';
+const static = 'white';
```
Comment (Rajdeepc) [inline packages/action-button/stories/action-button-white-quiet.stories.ts:23]:
```suggestion
const staticProp = 'white';
```

File: packages/action-button/stories/action-button-white-quiet.stories.ts [hunk 3]
```diff
@@ -20,40 +20,40 @@ export default {
...
-    variant,
+    static,
 };
 
 export const s = (args: Properties): TemplateResult => renderButtons(args);
 s.args = {
     size: 's',
     quiet,
-    variant,
+    static,
```
Comment (Rajdeepc) [inline packages/action-button/stories/action-button-white-quiet.stories.ts:37]:
```suggestion
    staticProp,
```

File: packages/action-button/stories/action-button-white-quiet.stories.ts [hunk 4]
```diff
@@ -20,40 +20,40 @@ export default {
...
 };
 
 export const m = (args: Properties): TemplateResult => renderButtons(args);
 m.args = {
     size: 'm',
     quiet,
-    variant,
+    static,
```
Comment (Rajdeepc) [inline packages/action-button/stories/action-button-white-quiet.stories.ts:44]:
```suggestion
    staticProp,
```

File: packages/action-button/stories/action-button-white-quiet.stories.ts [hunk 5]
```diff
@@ -20,40 +20,40 @@ export default {
...
 };
 
 export const l = (args: Properties): TemplateResult => renderButtons(args);
 l.args = {
     size: 'l',
     quiet,
-    variant,
+    static,
```
Comment (Rajdeepc) [inline packages/action-button/stories/action-button-white-quiet.stories.ts:51]:
```suggestion
    staticProp,
```

File: packages/action-button/stories/action-button-white-quiet.stories.ts [hunk 6]
```diff
@@ -20,40 +20,40 @@ export default {
...
 };
 
 export const XL = (args: Properties): TemplateResult => renderButtons(args);
 XL.args = {
     size: 'xl',
     quiet,
-    variant,
+    static,
```
Comment (Rajdeepc) [inline packages/action-button/stories/action-button-white-quiet.stories.ts:58]:
```suggestion
    staticProp,
```

File: packages/action-button/test/action-button.test.ts [hunk 7]
```diff
@@ -282,76 +282,4 @@ describe('ActionButton', () => {
...
-            expect(spyCall.args.at(-1), 'confirm `data` shape').to.deep.equal({
-                data: {
-                    localName: 'sp-action-button',
-                    type: 'api',
-                    level: 'deprecation',
-                },
-            });
-        });
-
-        it('prefers `staticColor` over `static`', async () => {
```
Comment (Rajdeepc) [inline packages/action-button/test/action-button.test.ts:343]: we need to keep a test for staticColor

**Review by Rajdeepc — CHANGES_REQUESTED** [review]

File: packages/action-button/test/action-button.test.ts (hunk 7 above)
Comment (rubencarvalho) [reply packages/action-button/test/action-button.test.ts:343]: added one!

**Review by Rajdeepc — APPROVED** [review]

**Pull request merged**